controller/instance_manager_controller.go

```diff
@@ -1037,6 +1039,21 @@ func (imc *InstanceManagerController) createInstanceManagerPod(im *longhorn.Inst
 		return err
 	}
 
+	storageNetwork, err := imc.ds.GetSetting(types.SettingNameStorageNetwork)
+	if err != nil {
+		return err
+	}
+
+	nadAnnot := string(types.CNIAnnotationNetworks)
+	if storageNetwork.Value != types.CniNetworkNone {
+		err := types.ValidateStorageNetwork(storageNetwork.Value)
+		if err != nil {
+			return errors.Wrap(err, "failed to validate network-attachment-definition to create instance manager pod")
+		}
+
+		podSpec.Annotations[nadAnnot] = types.CreateCniAnnotationFromSetting(storageNetwork)
```

Comment:
Will there be any impacts if users add similar annotations manually for the existing pods? I think NO, right?

Reply:
The annotation will only take effect when the pod is created, it has no effect on running pods.